shared/AppInsightsCore/src/JavaScriptSDK/AppInsightsCore.ts

```diff
@@ -647,6 +535,124 @@ export class AppInsightsCore<CfgType extends IConfiguration = IConfiguration> im
                 return _startLogPoller(true);
             };
 
+            function _handleIKeyEndpointPromises(theConfig: IConfiguration) {
+                // app Insights core only handle ikey and endpointurl, aisku will handle cs
+                // But we want to reference these config values so that if any future changes are made
+                // this will trigger the re-run of the watch function
+                // and the ikey and endpointUrl will be set to the new values
+                let ikey = theConfig.instrumentationKey;
```

Comment:
These 2 lines are really the key part of the referencing change, as per the comment if these direct configuration properties are not referenced as part of the subsequent call to the onConfigChange (they where originally after the "pending" check) then the dynamic configuration change usage detection doesn't identify that this function should be called (indirectly by the calling onConfigChange (watch) function) again if / when either of these configuration options change.

Reply:
Apart from moving these 2 referenced variables to before the pending check (and not having a `isPending` variable) the reset of the logic of this function "should" (please check) be the same.

Reply:
The initial design of setting ikey after pending check is to avoid the scenarios where user changes the ikey to a string value before the previous ikey promises are resolved. But this did block the following onConfigChange before all promises are resolved.